melodies_monet/driver.py

```diff
@@ -148,41 +148,76 @@ def open_obs(self, time_interval=None):
+                    self.obj = xr.open_dataset(self.file)
+                elif extension in ['.ict', '.icarrt']:
+                    self.obj = mio.icarrt.add_data(self.file)
+                self.mask_and_scale()  # mask and scale values from the control values
+        except ValueError:
+            print('something happened opening file')
+            
+    def open_sat_obs(self,time_interval=None):
+        """Methods to opens satellite data observations. 
+        Uses in-house python code to open and load observations.
+        Alternatively may use the satpy reader.
 
-        self.mask_and_scale()  # mask and scale values from the control values
-        self.filter_obs()
+        Returns
+        -------
+        type
+            Fills the object class associated with the equivalent label (self.label) with satellite observation
+            dataset read in from the associated file (self.file) by the satellite file reader
+        """
+        from .util import time_interval_subset as tsub
+        
+        try:
+            if self.label == 'omps_l3':
+                self.obj = mio.sat._omps_l3_mm.read_OMPS_l3(self.file)
+            elif self.label == 'omps_nm':
+                if time_interval is not None:
+                    flst = tsub.subset_OMPS_l2(self.file,time_interval)
+                else: flst = self.file
+
+                self.obj = mio.sat._omps_nadir_mm.read_OMPS_nm(flst)
+                
+                # couple of changes to move to reader
+                self.obj = self.obj.swap_dims({'x':'time'}) # indexing needs
+                self.obj = self.obj.sortby('time') # enforce time in order. 
+                # restrict observation data to time_interval if using
+                # additional development to deal with files crossing intervals needed (eg situtations where orbit start at 23hrs, ends next day).
+                if time_interval is not None:
+                    self.obj = self.obj.sel(time=slice(time_interval[0],time_interval[-1]))
+                    
+            elif self.label == 'mopitt_l3':
+                print('Reading MOPITT')
+                self.obj = mio.sat._mopitt_l3_mm.read_mopittdataset(self.file, 'column')
+            elif self.label == 'modis_l2':
+                from monetio import modis_l2
+                print('Reading MODIS L2')
+                self.obj = modis_l2.read_mfdataset(
+                    self.file, self.variable_dict, debug=self.debug)
+            elif self.label == 'tropomi_l2_no2':
+                from monetio import tropomi_l2_no2
+                print('Reading TROPOMI L2 NO2')
+                self.obj = tropomi_l2_no2.read_trpdataset(
+                    self.file, self.variable_dict, debug=self.debug)
+            else: print('file reader not implemented for {} observation'.format(self.label))
+        except ValueError:
+            print('something happened opening file')
```

Comment:
It would be better not to use `try` block and just let possible exception be raised, so that you get info about what the something that happened is.